libs/sdk-bindings/src/breez_sdk.udl

```diff
@@ -435,6 +445,24 @@ dictionary ReverseSwapFeesRequest {
     u32? claim_tx_feerate = null;
 };
 
+dictionary FetchOnchainLimitsResponse {
+    u64 min_sat;
+    u64 max_sat;
+    string fees_hash;
```

Comment:
I think it complicates the flow if we return here fee hash and require it to be passed to the PrepareOnchain.
Seems simpler to detach the limits call from the prepare call.
Given the fact they are being called almost together the probability for limits change that affect the user amount seems very small.

Reply:
This strongly depends on how the app handles this:
- call `onchain_payment_limits` once on app startup, then always use same limits
- call `onchain_payment_limits` when needed, but cache results
- call `onchain_payment_limits` for each new swap
- call `onchain_payment_limits` for each new swap and when `FeePromiseChanged` error is thrown

IMO the first two are much more likely than the last two.

As a consequence, they're more likely to face the changed limits situation. Having the `fees_hash` in the `onchain_payment_limits` result prevents this.

I can remove it, but because of the above, doing so will IMO open the door to `OutOfRange` errors while the user input appears to be in range.

Should I still remove it?

Reply:
@ok300  Even if we leave it we still create a new fee_hash in prepareOnchain so we can't use it anyway...

Reply:
Actually, the fresh `fee_hash` in `prepare` allows us to check if the `onchain_payment_limits` `fee_hash` is expired or not.

So, by leaving it, we can do this check. By removing it, we expose the caller to the risks above.

Reply:
That last commit that compares the input fee_hash to the new fee_hash takes into consideration some non validated assumptions on the swapper internal logic. We don't know how they generate these hashes and if they decide to return new hash for every request based on time or some other factors (even though the limit hasn't change).
So this has risk IMO and I strongly think that for simplicity we should remove the fee_hash from the onchain limits.

Reply:
> We don't know how they generate these hashes

We do know:

> `hash`: SHA256 hash of the JSON encoded data in the pair object
> https://docs.boltz.exchange/v/api/api-v1#supported-pairs

The pair object contains only the fees and the limits, see https://api.boltz.exchange/getpairs , no nonce or timestamp-related value.

> some non validated assumptions on the swapper internal logic

In addition to the docs, this has been validated by the current v1 reverse swap. The fact that that `pay` in v1 asks for the `fees_hash` and uses it as arg in the create request, means this `fees_hash` must be valid. Otherwise, `pay` would fail. If however `getpairs` shows a different hash, and we now call `pay` with a slightly older hash, it will also fail. So the hash remains the same for as long as the fee promise is valid.

Despite all this, should I
- remove this validation from `pay`
- remove the `fees_hash` field from `onchain_payment_limits` response + from the `prepare` arg + remove this validation from `prepare`
?

Reply:
> This strongly depends on how the app handles this:
> 
> * call `onchain_payment_limits` once on app startup, then always use same limits
> * call `onchain_payment_limits` when needed, but cache results
> * call `onchain_payment_limits` for each new swap
> * call `onchain_payment_limits` for each new swap and when `FeePromiseChanged` error is thrown

I think the 3rd / 4th is more likely, there is no point caching it as it's rarely used and can be an initialisation function of a Swap component

> Despite all this, should I
> 
> * remove this validation from `pay`
> * remove the `fees_hash` field from `onchain_payment_limits` response + from the `prepare` arg + remove this validation from `prepare`
>   ?

I would say the 2nd. For me it doesn't make sense to keep hold of the hash as it's not a promise (or did I get this wrong?). The SDK could also check the amount is within the current limits without the hash, then return the `prepare` response with fees and hash. Having the hash here (and passed to `pay`) is a fair indicator the app has shown the fees to the user and to use them when calling `pay`

Reply:
I agree with Ross. I think we should:
1. remove the fee_hash from the onchain_payment_limits
2. in pay_onchain, create the swap (no need to call fetch_fees again) and then validate the amounts with the returned data. No need to validate that the hash is the same, boltz already does it when we create the swap.